qa/tasks/cephfs/test_cephfs_shell.py

```diff
@@ -175,6 +175,111 @@ def test_mkdir_path_with_path_option(self):
         o = self.mount_a.stat('d5/d6/d7')
         log.info("mount_a output:\n{}".format(o))
 
+class TestRMdir(TestCephFSShell):
+    dir_name = "test_dir"
+
+    def is_dir_deleted(self, msg, opt=" "):
+        self.run_cephfs_shell_cmd("rmdir"+ opt + self.dir_name)
+        self.assertFalse(path.exists(path.join(self.mount_a.mountpoint,
+                                               self.dir_name)), msg)
+    """
+    def dir_not_deleted(self, msg, opt=" "):
+        rmdir_output = self.get_cephfs_shell_cmd_error("rmdir" + opt + self.dir_name)
+        log.info("cephfs-shell rmdir output:\n{}".format(rmdir_output))
+        self.assertTrue(path.exists(path.join(self.mount_a.mountpoint,
+                                              self.dir_name)), msg)
+    """
+    def test_rmdir(self):
+        """
+        Test that rmdir deletes directory
+        """
+        self.run_cephfs_shell_cmd("mkdir " + self.dir_name)
+        self.is_dir_deleted("test_dir is not deleted")
+
+    def test_rmdir_non_existing_dir(self):
+        """
+        Test that rmdir outputs error for non existing directory
+        """
+        try:
+            rmdir_output = self.get_cephfs_shell_cmd_error("rmdir test_dir")
+        except Exception as e:
+            log.info("cephfs-shell rmdir Exception:\n{}".format(e))
+            pass
+
+        self.assertTrue(rmdir_output, "Something went wrong!! non existing dir deleted")
+
+    def test_rmdir_non_empty_dir(self):
+        """
+        Test that rmdir outputs error for non empty directory
+        """
+        self.run_cephfs_shell_cmd("mkdir " + self.dir_name)
+        self.run_cephfs_shell_cmd("put - test_dir/dumpfile", stdin="Valid File")
+        try:
+            self.run_cephfs_shell_cmd("rmdir" + self.dir_name)
+        except Exception as e:
+            log.info("cephfs-shell rmdir Exception:\n{}".format(e))
+            pass
+
+        self.assertTrue(path.exists(path.join(self.mount_a.mountpoint,
+                        self.dir_name)), "non-empty test_dir is deleted")
+
+    def test_rmdir_existing_file(self):
+        """
+        Test that rmdir outputs error on file
+        """
+        self.run_cephfs_shell_cmd("put - dumpfile", stdin="Valid File")
+        try:
+            self.run_cephfs_shell_cmd("rmdir dumpfile")
+        except Exception as e:
+            log.info("cephfs-shell output Exception:\n{}".format(e))
+            pass
+
+        self.assertTrue(path.exists(path.join(self.mount_a.mountpoint, "dumpfile")),
+                        "Something went wrong!! rmdir deleted a file")
```

Comment:
The reason this works with `vstart_runner.py` and not teuthology is that `mount_a` mount is on the same machine as your the process executing these tests. When you run on teuthology, this python code is run separately from the ceph cluster. You need to execute a `stat` shell command or simliar on one of hte client mounts to check the existence of the file. See `stat` in mount.py.

Reply:
````
2020-01-13T09:56:30.261 INFO:tasks.cephfs_test_runner:======================================================================
2020-01-13T09:56:30.261 INFO:tasks.cephfs_test_runner:FAIL: test_rmdir_existing_file (tasks.cephfs.test_cephfs_shell.TestRMdir)
2020-01-13T09:56:30.261 INFO:tasks.cephfs_test_runner:----------------------------------------------------------------------
2020-01-13T09:56:30.261 INFO:tasks.cephfs_test_runner:Traceback (most recent call last):
2020-01-13T09:56:30.261 INFO:tasks.cephfs_test_runner:  File "/home/teuthworker/src/git.ceph.com_ceph-c_wip-cephfs-shell-fix-rmdir-p-option/qa/tasks/cephfs/test_cephfs_shell.py", line 238, in test_rmdir_existing_file
2020-01-13T09:56:30.262 INFO:tasks.cephfs_test_runner:    "Something went wrong!! rmdir deleted a file")
2020-01-13T09:56:30.262 INFO:tasks.cephfs_test_runner:AssertionError: Something went wrong!! rmdir deleted a file
````

Reply:
Thanks for explaining. I have updated the tests. Please review them again.